### PR TITLE
feat: implemented clusters creation

### PR DESCRIPTION
**Pull request by roolrd**
modified:   README.md
	new file:   cluster.tf
	modified:   main.tf
	modified:   outputs.tf
	new file:   permissions.tf
	modified:   secrets.tf
	modified:   variables.tf

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -1,67 +1,121 @@
...
+    name       = null
+    can_use    = null
+    definition = null
+  }]
+}
+
+variable "clusters" {
+  type = set(object({
+    cluster_name                 = string
+    spark_version                = optional(string, "11.3.x-scala2.12")
```
Comment (owlleg6) [inline variables.tf:95]: set at least 13+ version for runtime

**Review by owlleg6 — APPROVED** [review]

File: variables.tf [hunk 2]
```diff
@@ -1,67 +1,121 @@
...
+    spark_conf                   = optional(map(any), {})
+    cluster_conf_passthrought    = optional(bool, false)
+    spark_env_vars               = optional(map(any), {})
+    data_security_mode           = optional(string, "USER_ISOLATION")
+    node_type_id                 = optional(string, "")
+    autotermination_minutes      = optional(number, 30)
+    min_workers                  = optional(number, 1)
+    max_workers                  = optional(number, 2)
+    availability                 = optional(string, "PREEMPTIBLE_WITH_FALLBACK_GCP")
+    zone_id              = optional(string, "")
```
Comment (owlleg6) [inline variables.tf:105]: run terraform fmt

File: variables.tf (hunk 1 above)
Comment (roolrd) [reply variables.tf]: done

**Review by owlleg6 — APPROVED** [review]

**Review by owlleg6 — APPROVED** [review]
lgtm

**Pull request merged**